cli/command/container/cp.go

```diff
@@ -217,17 +224,16 @@ func copyFromContainer(ctx context.Context, dockerCli command.Cli, copyConfig cp
 		preArchive = archive.RebaseArchiveEntries(content, srcBase, srcInfo.RebaseName)
 	}
 
-	if !copyConfig.quiet {
-		fmt.Fprint(dockerCli.Out(), "Preparing to copy...\n")
-		fmt.Fprint(dockerCli.Out(), "\033[s")
+	if copyConfig.quiet {
+		return archive.CopyTo(preArchive, srcInfo, dstPath)
 	}
 
+	fmt.Fprint(dockerCli.Err(), "Preparing to copy...\n")
+	fmt.Fprint(dockerCli.Err(), "\033[s")
+	fmt.Fprint(dockerCli.Err(), aec.Up(1))
 	res := archive.CopyTo(preArchive, srcInfo, dstPath)
-
-	if !copyConfig.quiet {
-		fmt.Fprint(dockerCli.Out(), "\n")
-		fmt.Fprint(dockerCli.Out(), "Successfully copied ", units.HumanSize(float64(copiedSize)), " to ", dstPath, "\n")
-	}
+	fmt.Fprint(dockerCli.Err(), "\n")
```

Comment:
Changed all output to print on `STDERR` (perhaps needs to be discussed on the docker/cli PR)

Reply:
Uhm, I'm confusing with this change. Why we need change the output to `STDERR`?

Reply:
Uhm, I'm confusing with this change. Why we need change the output to `STDERR`?

Reply:
Oh, we will use `STDERR` only when `quiet` is true?

Reply:
There's slightly more details in https://github.com/docker/cli/pull/1859/files. Thinking of that, perhaps the `Successfully copied` output should be printed on `STDOUT`.

The general idea is that "informational" output generally goes to `STDERR`; this allows users to redirect output elsewhere.

For example;

```console
$ curl https://get.docker.com 2> progress.log | grep Desktop
		echo "WSL DETECTED: We recommend using Docker Desktop for Windows."
		echo "Please get Docker Desktop from https://www.docker.com/products/docker-desktop"
					echo "Please get Docker Desktop from https://www.docker.com/products/docker-desktop"
```

In the above, informational output (progress bars) of curl are printed on `STDERR` and can be redirected (in this case I'm sending it to `progress.log`), but the output of the file that's downloaded is still on `STDERR`, and I can use `grep` to search in the content of that file.

Reply:
Uhm, ok, I understand that, nice.